crates/evm/traces/src/decoder/mod.rs

```diff
@@ -327,10 +329,23 @@ impl CallTraceDecoder {
     pub async fn populate_traces(&self, traces: &mut Vec<CallTraceNode>) {
         for node in traces {
             node.trace.decoded = self.decode_function(&node.trace).await;
+
             for log in node.logs.iter_mut() {
                 log.decoded = self.decode_event(&log.raw_log).await;
             }
 
+            if let Ok(e) = Vm::UnemittedEventError::abi_decode(&node.trace.output, false) {
```

Comment:
I could actually add the unmatched event name to the revert error here, if that's desirable.